lib/mutant/cache.rb

```diff
@@ -21,7 +24,7 @@ def initialize
     # @api private
     def parse(path)
       @cache.fetch(path) do
-        @cache[path] = Parser::CurrentRuby.parse(File.read(path))
+        @cache[path] = Parser::CurrentRuby.parse(file.read(path))
```

Comment:
Also if you use `ThreadSafe::Cache` you can use it's `#fetch_or_store` method to simplify #parse:

```ruby
def initialize(*)
  super
  @cache = ThreadSafe::Cache.new
end

# ...

def parse(path)
  @cache.fetch_or_store(path) do |path|
    Parser::CurrentRuby.parse(file.read(path))
  end
end
```